src/openassetio-python/package/openassetio/pluginSystem/PythonPluginSystem.py

```diff
@@ -257,11 +258,10 @@ def __load(self, path):
 
         self.register(module.plugin, path)
 
-    def __log_exc(self, ex):
+    @staticmethod
+    def __format_exc(ex):
         """
-        Logs information about the supplied exception into the logger
-        with appropriate formatting.
+        Provides a multi-line string with information about the supplied
+        exception with appropriate formatting.
         """
-        for frame in traceback.format_tb(ex.__traceback__):
-            for line in frame.splitlines():
-                self.__logger.error(line)
+        return "/n".join(traceback.format_tb(ex.__traceback__))
```

Comment:
[consider] I'm a little wary of directly accessing magic attributes. I think `traceback.format_exc()` works for us, since we're always printing the exception during the stack unroll?

Reply:
Yeah I was wondering about this, but it is explicitly documented [here](https://docs.python.org/3/reference/datamodel.html?highlight=__traceback__):

> [Traceback objects] ... It is accessible as the third item of the tuple returned by sys.exc_info(), and as the __traceback__ attribute of the caught exception.

I didn't like a helper function assuming it was in the stack unroll, and I didn't like duplicating how we handled the exceptions, so this felt reasonable.

Reply:
Since `traceback.format_exc()` gives us a string, there would be no need for a helper function.  But yeah it is odd and annoying that there is no non-stateful way to do this.  Feel free to resolve this and squash+merge if you want.

Reply:
Ahh sorry, id been looking at `traceback.format_exception()` which needs `sys.last...` 🙃 . Much [simpler](https://github.com/OpenAssetIO/OpenAssetIO/pull/768/commits/bf873888c5796e6222d268880147f716173ea790). Thanks.